### PR TITLE
Adding stooge sort

### PR DESCRIPTION
**Pull request by Charitoc**

### CODE REVIEW

**Comment by cclauss** [comment]
The failing Travis CI builds are not caused by this PR but are instead caused by my mistake in #1205.

File: sorts/stooge_sort.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+def stooge(arr, i, h):
+
+    """
+
+>>> arr = [2, 4, 5, 3, 1]
+>>> stoogesort(arr)
+>>> print(arr)
+[1, 2, 3, 4, 5]
+
+"""
```
Comment (cclauss) [inline sorts/stooge_sort.py]: The doctest for __stooge_sort()__ should be in the __stooge_sort()__ function, not in the __stooge()__ function.

Let's move lines 6, 8, 9, 10, 11, and 13 up to just below line 1.  The blank lines 7 and 12 can be deleted.  It should look like:
```python
def stooge_sort(arr):
    """
    >>> arr = [2, 4, 5, 3, 1]
    >>> stoogesort(arr)
    >>> print(arr)
    [1, 2, 3, 4, 5]
    """
    stooge(arr, 0, len(arr) - 1)


def stooge(arr, i, h):
```

Comment (Charitoc) [reply sorts/stooge_sort.py]: Done

**Comment by cclauss** [comment]
While waiting, you should be able to test this locally with __python3 -m doctest -v sorts/stooge_sort.py__

**Comment by Charitoc** [comment]
It is ready, could you do the review?

**Comment by Charitoc** [comment]
Thank you!

**Comment by cclauss** [comment]
On thing to think about in the future is the difference between __list.sort()__ and __sorted(list)__ in Python.  The first is an in-place sort (like this stooge_sort()) with the second returns a new sorted list without modifying the original list.  Both are useful but I tend to like the second better.

https://docs.python.org/3/howto/sorting.html

**Comment by Charitoc** [comment]
Ok, thank you for your time.

**Pull request merged**